### DISCUSSION TITLE
Is FormData encoded by default?

### DISCUSSION THREAD

**Discussion by michealroberts** [go-resty/resty#890] [Q&A] [answered]
I'm just digging into a bug in a library I have created, and I was wondering if I need to encode FormData before sending, e.g.:

```go
r := a.Client.R().SetHeader("Content-Type", "application/x-www-form-urlencoded").SetResult(&putResponse{}).SetHeader("Accept", "application/json").SetFormData(form)

r.FormData.Encode()

resp, err := r.Put(url)

if err != nil {
	return err
}
```

or if this is done automatically?

I can't see anywhere in the code where it is done automatically, but I just wanted confirmation from a library maintainer/author before assuming this is the case ...

Many thanks! 😃

**Comment by jeevatkm** [comment, marked as answer]
@michealroberts Thanks for reaching out. 

Resty does the Form data encoding and sets appropriate content type when request form data methods are used. You could use the search string `SetForm` on Resty godoc to find those methods.

Code reference:
https://github.com/go-resty/resty/blob/a6a489ba12058a87e4e2ed4aace5190e9184c93d/middleware.go#L501-L502

**Comment by michealroberts** [comment]
@jeevatkm Many thanks for your response, I appreciate that. I searched Encode() but I wasn't sure if it needed to be explicitly / specifically called.
